backends/ent/retrieve.go

```diff
@@ -111,3 +112,26 @@ func (backend *Backend) GetExternalReferencesByDocumentID(
 
 	return extRefs, nil
 }
+
+func (backend *Backend) GetNodesByID(ids ...string) ([]*sbom.Node, error) {
+	nodes := []*sbom.Node{}
+
+	predicates := []predicate.Node{}
+
+	if len(ids) > 0 {
+		predicates = append(predicates, node.IDIn(ids...))
+	}
+
+	results, err := backend.client.Node.Query().
+		Where(predicates...).
+		All(backend.ctx)
+	if err != nil {
+		return nil, fmt.Errorf("querying nodes table: %w", err)
+	}
+
+	for idx := range results {
+		nodes = append(nodes, results[idx].ProtoMessage)
+	}
```

Comment:
Try something like this to leverage the `sliceutil` package (double check for correct usage and syntax)

```go
	nodes := sliceutil.Extract(results, func(node *ent.Node) *sbom.Node {
		return node.ProtoMessage
	})
```